src/sentry/static/sentry/app/views/settings/project/projectFilters/projectFiltersChart.tsx

```diff
@@ -65,43 +60,35 @@ class ProjectFiltersChart extends React.Component<Props, State> {
     }
   }
 
-  getStatOpts() {
-    return {
-      'ip-address': 'IP Address',
-      'release-version': 'Release',
-      'error-message': 'Error Message',
-      'browser-extensions': 'Browser Extension',
-      'legacy-browsers': 'Legacy Browser',
-      localhost: 'Localhost',
-      'web-crawlers': 'Web Crawler',
-      'invalid-csp': 'Invalid CSP',
-      cors: 'CORS',
-      'discarded-hash': 'Discarded Issue',
-    };
-  }
-
-  formatData(rawData) {
-    return Object.keys(this.getStatOpts()).map(stat => ({
-      data: rawData[stat].map(([x, y]) => {
-        if (y > 0) {
+  formatData(rawData: RawStats) {
+    const seriesWithData: Set<string> = new Set();
+    const transformed = Object.keys(STAT_OPS).map(stat => ({
+      data: rawData[stat].map(([timestamp, value]) => {
+        if (value > 0) {
+          seriesWithData.add(STAT_OPS[stat].title);
           this.setState({blankStats: false});
         }
-
-        return {x, y};
+        return {name: timestamp * 1000, value};
       }),
-      label: this.getStatOpts()[stat],
-      statName: stat,
+      seriesName: STAT_OPS[stat].title,
+      color: STAT_OPS[stat].color,
     }));
+
+    return transformed.filter((series: Series) => seriesWithData.has(series.seriesName));
   }
 
   getFilterStats() {
-    const statOptions = Object.keys(this.getStatOpts());
+    const statOptions = Object.keys(STAT_OPS);
     const {project} = this.props;
     const {orgId} = this.props.params;
+
+    const until = Math.floor(new Date().getTime() / 1000);
```

Comment:
I guess the only change moving this is that the getTime is updated every time this is rendered instead of set the first time in the constructor? So all re-renders will move this time forward, I don't know enough about the surrounding components or updates to be sure it will matter though.

Reply:
Yes that is correct, but I don't think that change will matter in practice as the bucket window for this chart is 1 day.